### PR TITLE
Add "no suggests" R CMD check workflow

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
This can help us catch instances of soft dependency usage in examples or in tests without the necessary skip.

### CODE REVIEW

File: .github/workflows/check-no-suggests.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# NOTE: This workflow only directly installs "hard" dependencies, i.e. Depends,
+# Imports, and LinkingTo dependencies. Notably, Suggests dependencies are never
+# installed, with the exception of testthat, knitr, and rmarkdown. The cache is
+# never used to avoid accidentally restoring a cache containing a suggested
+# dependency.
+#
+# Customization for `{lintr}`
+#
+# The following soft dependencies are not optional for testing:
+# `{patrick}`, `{rex}`, `{withr}`
```
Comment (AshesITR) [inline .github/workflows/check-no-suggests.yaml:13]: Why is rex a soft dep? We use it in our package code, no?

Comment (MichaelChirico) [reply .github/workflows/check-no-suggests.yaml:13]: shall we delete rex from the loadNamespace calls? I believe it was added by mistake

Comment (IndrajeetPatil) [reply .github/workflows/check-no-suggests.yaml]: Yeah, I think it was introduced accidentally. Made a separate PR to fix that.

**Comment by IndrajeetPatil** [comment]
We will need to keep this PR updated and pending until all relevant issues to get the new check to succeed are fixed in separate PRs.

**Comment by IndrajeetPatil** [comment]
The new check is green now.

**Review by AshesITR — APPROVED** [review]
Nice work, thanks!

**Pull request merged**